SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -653,6 +656,237 @@ private void DisplayErrorMessage(string message)
                     MessageBoxButton.OK, MessageBoxImage.Error);
         }
 
+        private void LoginButton_Clicked(object sender, RoutedEventArgs e)
+        {
+            LabelEmptyFieldsError.Visibility = Visibility.Collapsed;
+            LabelUsernameError.Visibility = Visibility.Collapsed;
+            LabelPasswordError.Visibility = Visibility.Collapsed;
+
+            var users = userLoginData.GetAllData();
+            var usernameExists = false;
+            string pswHash = null;
+            string pswSalt = null;
+            string username = null;
+            string userID = null;            
+
+            foreach (var name in users)
+            {
+                var checkUsernameExists = (name.Username == UsernameTextBox.Text);
+                if (checkUsernameExists)
+                {
+                    userID = name.ID;
+                    pswHash = name.PasswordHash;
+                    pswSalt = name.Salt;
+                    username = name.Username;
+                    usernameExists = true;
+                }
+            }
+
+            UserData registeredUser = new UserData(username, pswHash, pswSalt, userID);
+            UserData offlineUser = new UserData(UsernameTextBox.Text, null, pswSalt, userID);
```

Comment:
Instead of creating a new user just to generate a hash, you could make the GenerateHash function static, and then just do 
string hash = UserData.GenerateHash(UsernameTextBox.Text, pswSalt);